src/lib/channels.py

```diff
@@ -0,0 +1,116 @@
+import time
+
+from core_data_modules.cleaners import Codes
+from core_data_modules.traced_data import Metadata
+from dateutil.parser import isoparse
+
+from src.lib import PipelineConfiguration
+
+class Channels(object):
+    RADIO_PROMO_KEY = "radio_promo"
+    RADIO_SHOW_KEY = "radio_show"
+    NON_LOGICAL_KEY = "non_logical_time"
+    S01E01_KEY = "radio_participation_s01e01"
+    S01E02_KEY = "radio_participation_s01e02"
+    S01E03_KEY = "radio_participation_s01e03"
+    S01E04_KEY = "radio_participation_s01e04"
+    S01E05_KEY = "radio_participation_s01e05"
+    S01E06_KEY = "radio_participation_s01e06"
+
+    # Time ranges expressed in format (start_of_range_inclusive, end_of_range_exclusive)
+    RADIO_PROMO_RANGES = [
+        ("2019-04-02T00:00:00+0300", "2019-04-04T23:59:59+0300"),
```

Comment:
If you do this then you're ignoring messages received at e.g. 2019-04-04T23:59:59.500+03:00.
The last end date is exclusive, so instead you can say 2019-04-04T24:00:00+03:00, which means anything before the end of day on April 4th.

Reply:
Great thanks for this.

Reply:
Have updated to ("2019-04-02T00:00:00+0300", "2019-04-05T07:00:00+0300"), To cover promos that may be aired early morning before the show.